File: typescript/resistor-color-trio/Iteration1/resistor-color-trio.ts

export function decodedResistorValue(colors: string[]): string {
  const resistorValue = decodedValue(colors);

  var value = resistorValue;
  var numZeros = 0;

  while (resistorValue > 0 && value % 10 == 0) {
    value /= 10;
    numZeros++;
  }
  
  switch (numZeros) {
    case 9: return `${resistorValue/1000000000} gigaohms`;
    case 6: return `${resistorValue/1000000} megaohms`;
    case 3:
    case 4:
      return `${resistorValue/1000} kiloohms`;
    default:
      return `${resistorValue} ohms`;
  }
}

function decodedValue(colors: string[]) {
  const value = colors
    .slice(0, 2)
    .reverse()
    .map((color, index) => COLORS.indexOf(color) * Math.pow(10, index))
    .reduce((accumulated, current) => accumulated + current, 0);

  const numZeros = COLORS.indexOf(colors[2]);

  return Math.pow(10, numZeros) * value;
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
]

File: csharp/resistor-color-trio/Iteration1/ResistorColorTrio.cs
using System;

public static class ResistorColorTrio
{
    public static string Label(string[] colors)
    {
        var value = (Value(colors[0]) * 10) + Value(colors[1]);
        var numZeros = Value(colors[2]);

        for (int i = 0; i < numZeros; i++)
            value *= 10;
        
        return value switch
        {
            > 1000 => $"{value / 1000} kiloohms",
            _ => $"{value} ohms"
        };
    }

    private static int Value(string color) =>
        color switch
        {
            "black" => 0,
            "brown" => 1,
            "red" => 2,
            "orange" => 3,
            "yellow" => 4,
            "green" => 5,
            "blue" => 6,
            "violet" => 7,
            "grey" => 8,
            "white" => 9
        };
}
